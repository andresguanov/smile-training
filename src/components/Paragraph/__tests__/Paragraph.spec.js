import { shallowMount, mount } from "@vue/test-utils"
import smParagraph from "@components/Paragraph/Paragraph"
let wrapper
const staticClass = "static-class"
const staticStyle = "font-size: 16px;"

describe("smParagraph", () => {
  //#region Basic render
  it("renders default smParagraph", () => {
    wrapper = shallowMount(smParagraph, {
      context: {
        props: {
          size: "small"
        },
        staticClass,
        staticStyle
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
    const classes = wrapper.classes()
    expect(classes).toContain("is-small")
    expect(classes).toContain(staticClass)
    expect(wrapper.attributes("style")).toBe(staticStyle)
    expect(wrapper.text()).toBe("")
  })
  //#endregion

  //#region Props
  //#endregion

  //#region Methods
  //#endregion

  //#region Events
  //#endregion

  //#region Slots
  it("actually renders content", () => {
    wrapper = shallowMount(smParagraph, {
      slots: {
        default: "default content"
      }
    })
    expect(wrapper.text()).toBe("default content")
  })
  //#endregion
})
