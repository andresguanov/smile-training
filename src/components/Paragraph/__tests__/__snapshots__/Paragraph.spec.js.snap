// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smParagraph renders default smParagraph 1`] = `<p class="is-small static-class" style="font-size: 16px;"></p>`;
